src/lib/mina_net2/libp2p_stream.ml

```diff
@@ -40,6 +40,7 @@ let data_received { incoming_w; _ } data =
   don't_wait_for (Pipe.write_if_open incoming_w data)
 
 let reset ~helper { id; _ } =
+  Libp2p_helper.log_rpc_request helper "ResetStream" ;
```

Comment:
It seems to make sense to me to just put all of these logging calls in `Libp2p_helper.do_rpc`. We should be able to associate the string name with the RPC request and log it inside of that logic.

Reply:
@nholland94 for `Publish` push message and `SendStream` RPC I also attach length of the data to the metadata. And for `Publish` and `Subscribe` a topic. This was the main reason why I decided to add these logs in this dumb way.

I probably could add `len` for every message and skip inclusion of topic and proceed in a unified way as you described. But I don't know an easy way to both do associations and size computation. If you have a good idea how to do it, may I ask for help here? Or just keep the dumb solution, which is fine by me too?

Reply:
In fact, we do have a log like you mentioned `[%log' spam t.logger] "sending $message_type to libp2p_helper"  ~metadata:[ ("message_type", `String Rpc.name) ] ;`

Let me just bump it from `spam` to `trace` and include message length. It's simpler in the morning than it looked like in the evening.